sklearnex/covariance/incremental_covariance.py

```diff
@@ -196,6 +198,43 @@ def _onedal_partial_fit(self, X, queue=None, check_input=True):
 
         return self
 
+    @wrap_output_data
+    def score(self, X_test, y=None):
+        xp, _ = get_namespace(X_test)
+
+        location = self.location_
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(
+                X_test,
+                dtype=[np.float64, np.float32],
+                reset=False,
+            )
+        else:
+            X = check_array(
+                X_test,
+                dtype=[np.float64, np.float32],
+            )
+
+        if "numpy" not in xp.__name__:
+            location = xp.asarray(location, device=X_test.device)
+            # depending on the sklearn version, check_array
+            # and validate_data will return only numpy arrays
+            # which will break dpnp/dpctl support. If the
+            # array namespace isn't from numpy and the data
+            # is now a numpy array, it has been validated and
+            # the original can be used.
+            if isinstance(X, np.ndarray):
+                X = X_test
+
+        est = clone(self)
+        est.set_params(**{"assume_centered": True})
```

Comment:
In sklearn, the calculation of the empirical covariance is a function, where the assumed_centered is a flag. It is a deeply hidden function which is re-used in this case to calculate the score with assumed_centered=True.   Since the oneDAL functionality isn't like this (an object with state), the only way to support non-numpy inputs and to change the assumed centered flag without overwriting the fit values was to use a sklearn copy of the estimator. Sort of a hack I'll admit.